File: python/meteroid/models/coupon.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .coupon_discount import CouponDiscount
from .coupon_id import CouponId
from .plan_id import PlanId


@dataclasses.dataclass
class Coupon(BaseModel):
    code: str

    created_at: datetime

    disabled: bool

    discount: CouponDiscount

    id: CouponId

    plan_ids: t.List[PlanId]

    redemption_count: int

    reusable: bool

    archived_at: t.Optional[datetime] = None

    description: t.Optional[str] = None

    expires_at: t.Optional[datetime] = None

    recurring_value: t.Optional[int] = None

    redemption_limit: t.Optional[int] = None

File: go/coupon.go
// this file is @generated
package meteroid

import "time"

type Coupon struct {
	// RFC 3339 timestamp.
	ArchivedAt *time.Time `json:"archived_at,omitempty"`

	Code string `json:"code"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	Disabled bool `json:"disabled"`

	Discount CouponDiscount `json:"discount"`

	// RFC 3339 timestamp.
	ExpiresAt *time.Time `json:"expires_at,omitempty"`

	Id CouponId `json:"id"`

	PlanIds RequiredSlice[PlanId] `json:"plan_ids"`

	RecurringValue *int32 `json:"recurring_value,omitempty"`

	RedemptionCount int32 `json:"redemption_count"`

	RedemptionLimit *int32 `json:"redemption_limit,omitempty"`

	Reusable bool `json:"reusable"`
}
